### ISSUE TITLE
postgreql proxy crashes

### ISSUE THREAD

**Issue by tetra12** [mholt/caddy-l4#264]
Hi I'm using Caddy-L4 to proxy `PostgreSQL` access. It works fine for some time (around 2 days) and then crashes. I have no idea how to debug this, just attaching an error report.

```sh
Nov 02 18:06:25 hyp0 caddy[1997329]:         /home/bku/work/go/pkg/mod/github.com/mholt/caddy-l4@v0.0.0-20241012124037-5764d700c21c/layer4/routes.go:156 +0x6>
Nov 02 18:06:25 hyp0 caddy[1997329]: github.com/mholt/caddy-l4/layer4.HandlerFunc.Handle(0x21c2138?, 0xc0008c0000?)
Nov 02 18:06:25 hyp0 caddy[1997329]:         /home/bku/work/go/pkg/mod/github.com/mholt/caddy-l4@v0.0.0-20241012124037-5764d700c21c/layer4/handlers.go:61 +0x>
Nov 02 18:06:25 hyp0 caddy[1997329]: github.com/mholt/caddy-l4/layer4.(*Server).handle(0xc000401260, {0x21c2138, 0xc0008c0000})
Nov 02 18:06:25 hyp0 caddy[1997329]:         /home/bku/work/go/pkg/mod/github.com/mholt/caddy-l4@v0.0.0-20241012124037-5764d700c21c/layer4/server.go:173 +0x1>
Nov 02 18:06:25 hyp0 caddy[1997329]: created by github.com/mholt/caddy-l4/layer4.(*Server).serve in goroutine 199
Nov 02 18:06:25 hyp0 caddy[1997329]:         /home/bku/work/go/pkg/mod/github.com/mholt/caddy-l4@v0.0.0-20241012124037-5764d700c21c/layer4/server.go:91 +0x27c
Nov 02 18:06:25 hyp0 systemd[1]: caddy.service: Main process exited, code=exited, status=2/INVALIDARGUMENT

```

Caddy: v2.8.4
Caddy-L4: v0.0.0-20241102143510-d8ba3fbdf35c
PostreSQL: 16.4
OS: Ubuntu 24.04.1

Caddyfile
```sh
{
  admin off
    layer4 {
    :55021 {
      @a postgres
      route @a {
        proxy {$DEV_KUBE_IP}:30000
      }
    }
  }
}
```

**Comment by vnxme** [comment]
@tetra12 Would you like to continue discussing this issue here, or has it been resolved in caddyserver/caddy#6674?

**Comment by tetra12** [comment]
@vnxme Yes, the issue seems to have been fixed. THanks for your feedback !
